packages/api/src/hooks/useDocumentUpload.ts

```diff
@@ -1,19 +1,78 @@
-import { useCallback, useMemo } from 'react';
+import { useCallback, useEffect, useMemo, useState } from 'react';
 import useMutation from '../useMutation';
+import { compressImageFile, generateChunks, numToUint8Array, combineBuffer, readFile } from '../utils';
+import md5 from 'md5';
+import { getWebsocket } from '../APIProvider';
 
 type TDocumentUploadPayload = Parameters<ReturnType<typeof useMutation<'document_upload'>>['mutate']>[0]['payload'];
+type TMutatePayload = TDocumentUploadPayload & {
+    file?: File;
+};
 
 /** A custom hook to handle document file uploads to our backend. */
 const useDocumentUpload = () => {
     const { data, mutate, ...rest } = useMutation('document_upload');
+    const [tempChunks, setTempChunks] = useState<Uint8Array[]>([]);
+    const derivAPI = getWebsocket();
+
+    const sendChunks = useCallback(
+        (chunks: Uint8Array[]) => {
+            const id = numToUint8Array(data?.document_upload?.upload_id || 0);
+            const type = numToUint8Array(data?.document_upload?.call_type || 0);
+
+            chunks.forEach(chunk => {
+                const size = numToUint8Array(chunk.length);
+                let payload = new Uint8Array([]);
+
+                payload = combineBuffer(payload, type);
+                payload = combineBuffer(payload, id);
+                payload = combineBuffer(payload, size);
+                payload = combineBuffer(payload, chunk);
```

Comment:
So we add identifiers to each frame 🤔  interesting. glad u find out a way

Reply:
I took it from library 🐱

Reply:
Siimplified it